src/platform_impl/macos/view.rs

```diff
@@ -644,54 +731,64 @@ extern "C" fn key_down(this: &Object, _sel: Sel, event: id) {
         let state_ptr: *mut c_void = *this.get_ivar("winitState");
         let state = &mut *(state_ptr as *mut ViewState);
         let window_id = WindowId(get_window_id(state.ns_window));
+
+        let input_source = current_input_source();
+        if state.input_source != input_source && state.is_ime_enabled() {
+            state.ime_state = ImeState::Disabled;
+            state.input_source = input_source;
+            AppState::queue_event(EventWrapper::StaticEvent(Event::WindowEvent {
+                window_id: WindowId(get_window_id(state.ns_window)),
+                event: WindowEvent::IME(IME::Disabled),
+            }));
+        }
```

Comment:
@kchibisov could you elaborate on the requirements on these?

Reply:
The purpose is that after you've got disabled you don't need to handle `set_ime_position` at all and if you had something in `preedit` you could safely discard it.

Reply:
Hmm, okay, but a `Preedit("", None, None)` could do the same, right?

Reply:
It means that you could have IME running, but just preedit is empty, so you still call `set_ime_postion` and potentially will call `set_surrounding_text`(if we add this one).

You could send empty preedit if you want though, but your preedit is empty anyway after the last `commit`, since this is how it usually works.